qutebrowser/misc/sessions.py

```diff
@@ -146,20 +146,13 @@ def _save_tab(self, tab, active):
                 encoded = item.originalUrl().toEncoded()
                 item_data['original-url'] = bytes(encoded).decode('ascii')
             user_data = item.userData()
+
             if history.currentItemIndex() == idx:
                 item_data['active'] = True
-                if user_data is None:
-                    pos = tab.page().mainFrame().scrollPosition()
-                    data['zoom'] = tab.zoomFactor()
-                    data['scroll-pos'] = {'x': pos.x(), 'y': pos.y()}
+                pos = tab.page().mainFrame().scrollPosition()
+                data['scroll-pos'] = {'x': pos.x(), 'y': pos.y()}
+                data['zoom'] = tab.zoomFactor()
             data['history'].append(item_data)
-
-            if user_data is not None:
-                if 'zoom' in user_data:
-                    data['zoom'] = user_data['zoom']
-                if 'scroll-pos' in user_data:
-                    pos = user_data['scroll-pos']
-                    data['scroll-pos'] = {'x': pos.x(), 'y': pos.y()}
```

Comment:
I think removing this would mean the zoom/scroll position for *older pages* in the history isn't saved anymore - though I can reproduce the bug you mentioned.

Currently, if you are on a page with 50% zoom, then load some other page and switch to 100% zoom there, saving the session, reloading and going back will show you the old page with 50% zoom.